h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -235,7 +235,14 @@ public static H2ONode self(InetAddress local) {
   }
 
   // Happy printable string
-  @Override public String toString() { return _key.toString (); }
+  @Override public String toString() {
+    String base = _key.toString();
```

Comment:
Did you check if we don't rely on the specific implementation of toString anywhere? Maybe toDebugString would be safer - but only if it is necessary.